### PR TITLE
New maps interface

### PR DESCRIPTION
**Pull request by papajohn**
I simplified the maps interface and changed the code structure. There's less functionality now, but I think that's a good thing. All regions are completely atomic — can't inspect them except to see a bit list of points. Markers are just markers without a circle radius. 

Though I changed a lot, the original implementation was immensely helpful.

See https://github.com/dsten/text/pull/70 for demos of the new interface. Can you take a look and make suggestions?

### CODE REVIEW

**Comment by papajohn** [comment]
Ok, I fixed a bunch of small issues. I think the interface is more stable now. I've made sure that @deculler 's Berkeley demo works again in the `map` branch of `text`.

File: tests/test_maps.py [hunk 1]
```diff
@@ -40,177 +16,136 @@ def region2(points):
...
 
-def test_draw_marker():
-    """ Tests that draw_map returns HTML """
-    points = [MapPoint((51.5135015, -0.1358392))]
-    map_html = draw_map((51.5135015, -0.1362392), points=points)
-    assert isinstance(map_html, HTML)
 
+def test_marker_html():
+    """ Tests that a Marker can be rendered. """
+    Marker(51.514, -0.132).show()
```
Comment (alvinwan) [inline tests/test_maps.py]: Wow.

File: tests/test_maps.py [hunk 2]
```diff
@@ -40,177 +16,136 @@ def region2(points):
...
+    Map([marker]).show()
 
-    point = MapPoint((51.5135015, -0.1358392))
-    map_html = draw_map(point)
-    assert isinstance(map_html, HTML)
 
+def test_map_region(states):
+    """ Tests that a Map can contain a Region. """
+    region = states['CA']
+    Map([region]).show()
```
Comment (alvinwan) [inline tests/test_maps.py]: This is really neat; could `Map` accept region/marker as a positional argument? i.e., wrap the `features` arg as a tuple if a subclass of `_FoliumWrapper`.

**Comment by alvinwan** [comment]
Cool, the syntax is definitely much more coder-friendly. LGTM. Feel free to merge.

File: tests/test_maps.py (hunk 2 above)
Comment (papajohn) [inline tests/test_maps.py]: Good idea. I'll do that.

**Pull request merged**